src/buffer/out/textBuffer.cpp

```diff
@@ -1165,7 +1165,7 @@ std::string TextBuffer::GenHTML(const TextAndColor& rows, const int fontHeightPo
                                 htmlBuilder << "&amp;";
                                 break;
                             default:
-                                htmlBuilder << ConvertToA(CP_UTF8, std::wstring_view(&c, 1));
+                                htmlBuilder << c;

```

Comment:
I'm still concerned about the cost of doing this character-by-character. I'd let everyone else on the team override me, though :smile:

Reply:
If I'm putting my Austin hat on, I'd say that there's certainly a `std` function that could be used here to just replace all these somehow. Maybe `std::replace` or `std::transform`?

That being said, I'm also a [third tribe](https://josephg.com/blog/3-tribes/) kinda guy and I think this is good enough. Someone else can choose to block this though, because yes we _should_ do things better. I just don't think this is a hot enough path to care ATM

Reply:
Honestly I think using std::replace or something like that is likely to introduce more intermediates and be of worse performance given that a single character is expanding into multiple in most of these situations.

Also, I do agree. This is a user-generated infrequent action, so I don't really care about premature optimization here. This makes sense as is and is easy to understand. I'm OK with it until a perf trace proves otherwise.